File: 01_Language/01_Functions/python/htmlentities.py
# coding: utf-8

from html import entities


def htmlentities(string):
    result = ""
    for s in string:
        if s == "&":
            result += "&amp;"
        elif s == "<":
            result += "&lt;"
        elif s == ">":
            result += "&gt;"
        elif s == '"':
            result += "&quot;"
        elif s == "'":
            result += "&#x27;"
        elif ord(s) in entities.codepoint2name:
            result += "&" + entities.codepoint2name[ord(s)] + ";"
        else:
            result += s
    return result


if __name__ == '__main__':
    print(htmlentities("A 'quote' is <b>bold中国</b>"))
    print(htmlentities("<Il était une fois un être>."))

File: 01_Language/01_Functions/php/htmlentities.php
<?php

$str = "A 'quote' is <b>bold</b>";

// Outputs: A 'quote' is &lt;b&gt;bold&lt;/b&gt;
echo htmlentities($str);
echo "\n";
// Outputs: A &#039;quote&#039; is &lt;b&gt;bold&lt;/b&gt;
echo htmlentities($str, ENT_QUOTES);
echo "\n";

echo htmlentities('<Il était une fois un être>.');
// Output: &lt;Il &eacute;tait une fois un &ecirc;tre&gt;.
//                ^^^^^^^^                 ^^^^^^^
echo "\n";

echo htmlspecialchars('<Il était une fois un être>.');
// Output: &lt;Il était une fois un être&gt;.
//
